src/categorical_algebra/FinCats.jl

```diff
@@ -104,17 +103,81 @@ is_discrete(C::FinCat) = isempty(hom_generators(C))
 """
 is_free(C::FinCat) = isempty(equations(C))
 
+"""Graph underlying a finitely presented category whose 
+object and
+hom generators are indexable, other
+than one explicitly generated by a graph.
+"""
+function graph(C::FinCat;maps::Bool=false,inv::Bool=false,named::Bool=false) 
+  g =  DiagramGraph() 
+  obgens = ob_generators(C)
+  homgens = hom_generators(C)
+  add_vertices!(g,length(obgens);vname=nameof.(obgens))
+  s(f) = only(indexin([dom(f)],obgens))
+  t(f) = only(indexin([codom(f)],obgens))
+  add_edges!(g,Int[s(f) for f in homgens],Int[t(f) for f in homgens];ename=nameof.(homgens)) 
+  named || begin g′ = Graph(); copy_parts!(g′,g,(:V,:E)); g = g′ end
+  maps || return g
+  inv ? (g, Dict(obgens[i] => i for i in 1:length(obgens)),
+  Dict(homgens[i] => i for i in 1:length(homgens))) :
+  (g, Dict(pairs(obgens)), Dict(pairs(homgens)))
+end
+#probably move into Graphs
+@present SchNamedGraph <: SchGraph begin
+  VName::AttrType
+  EName::AttrType
+  vname::Attr(V, VName)
+  ename::Attr(E, EName)
+end
+
+""" Abstract type for graph with named vertices and edges.
+"""
+@abstract_acset_type AbstractNamedGraph <: AbstractGraph
+
+""" Graph with named vertices and edges.
+
+The default graph type used by [`@graph`](@ref), [`@fincat`](@ref),
+[`@diagram`](@ref), and related macros.
+"""
+@acset_type NamedGraph(SchNamedGraph, index=[:src,:tgt,:ename],
+                       unique_index=[:vname]) <: AbstractNamedGraph
+vertex_name(g::AbstractNamedGraph, args...) = subpart(g, args..., :vname)
+edge_name(g::AbstractNamedGraph, args...) = subpart(g, args..., :ename)
+
+vertex_named(g::AbstractNamedGraph, name) = only(incident(g, name, :vname))
+edge_named(g::AbstractNamedGraph, name)= only(incident(g, name, :ename))
+const DiagramGraph = NamedGraph{Symbol,Symbol}
+
+
 # Opposite FinCats
 #-----------------
 
 const OppositeFinCat{Ob,Hom} = OppositeCat{Ob,Hom,FinCatSize}
 
 ob_generators(C::OppositeFinCat) = ob_generators(C.cat)
+#wrong for presentations, redefined below
 hom_generators(C::OppositeFinCat) = hom_generators(C.cat)
 
 ob_generator(C::OppositeCat, x) = ob_generator(C.cat, x)
 hom_generator(C::OppositeCat, f) = hom_generator(C.cat, f)
 
+"""
+Reverses the domain and codomain of certain types in a presentation,
+for use in an opposite category.
+"""
+function Base.reverse(p::Presentation,types::Vector{Symbol})
```

Comment:
Since this function is really about categories and uses the functions `dom` and `codom`, hence is not something that makes sense on a generic presentation, would suggest that this function not overload `Base.reverse` and be called something like `reverse_dom_codom`.